### ISSUE TITLE
I'm having a problem building Zim on Windows 8.1 64-bit

### ISSUE THREAD

**Issue by galiloe** [zim-desktop-wiki/zim-desktop-wiki#1269]
Hi,

I've followed the steps given on 

https://github.com/zim-desktop-wiki/zim-desktop-wiki/tree/master/windows

Everything seems to go well  until I run the ./build.sh. Then for some reason the Pyinstaller process creates a very large ANALYSIS-00.txt file that is referencing a load of files from C:\ProgramData (including some security protected files). I can't see why it should be referencing these at all, and it causes the process to crash when trying to copy these as part of building the COLLECT-00.toc. 

I've looked at the .toc files, warn-zim.txt, and the xref-zim.html files but I cannot see any obvious explanation. Given that the new Zim 0-73.2 windows installer works perfectly, I must be doing something wrong.

Any advice would be appreciated.

**Comment by jaap-karssenberg** [comment]
Yes there is a bug with latest version of pyinstaller. I copied the
reference in another report.

Op do 24 sep. 2020 16:35 schreef galiloe <notifications@github.com>:

**Comment by galiloe** [comment]
Thanks for the info. The problem was leaving me totally baffled. Thank you and your team for all the work in creating the new installer.

**Comment by galiloe** [comment]
I've now tried running ./build.sh with the version of pyinstaller set to 3.6. This successfully builds the installer. I can use this to install Zim to Windows, but the program refuses to start. I've gone through the various install directories - comparing files to the list installed by the official installer, and everything looks the same, apart from libssp-0.dll appearing in my installer.

**Comment by jaap-karssenberg** [comment]
With the latest version, zim should log all errors to "zim.log" file in the
tmp folder. Have a look at %TMP% - the exact subfolder is a random key.
If this log isn't there, issues occurs before zim code gets started.

Also might be useful to see whether zim runs currently from source under
the msys environment.

-- Jaap

**Comment by galiloe** [comment]
Thanks for your help. I've looked in %TMP% as you suggested. There are logs for when I installed and ran the official version, but I get no log when I try to run the version I have built (I downloaded the 0.72.3 release version of the code from github).

Running Zim with python3 zim.py within the mysys environment (a MSYS minGW 64-bit console) works perfectly. I can even use the spellcheck as I've installed the dependencies for that in msys.

**Comment by galiloe** [comment]
An update. I looked down the temp folder and found **zim.exe.log**.  This had the following error information

Traceback (most recent call last):
  File "gi/importer.py", line 137, in load_module
  File "gi/module.py", line 257, in get_introspection_module
  File "gi/module.py", line 109, in __init__
gi.RepositoryError: Typelib file for namespace 'HarfBuzz', version '0.0' not found

I've checked in MSYS and the file HarfBuzz-0.0.typelib is in the girepository, but hasn't been included in my installer. I copied this file to the gi_typelibs folder of the install from my build and Zim now works. 

Just have to figure out why this file wasn't put in the installer.  I googled this and there had been a problem with this with pyinstaller (https://github.com/pyinstaller/pyinstaller/issues/5129), but that had apparently been fixed.

**Comment by galiloe** [comment]
I've explored further. There have been changes to GTK3 so that HarfBuzz is now referenced. The development version of Pyinstaller has had a hook added to deal with this: hook-gi.repository.HarfBuzz.py. This was done 23 days ago. I need to find out how to use the development version in the Zim build script.

**Comment by galiloe** [comment]
I've now got things working. In the end I stayed with using version 3.6 of pyinstaller, as the development version still has the problem I reported at the start of this thread. Instead I modified the zim.spec file and added the following line to the datas parameter in Analysis

('../../../mingw64/lib/girepository-1.0/HarfBuzz-0.0.typelib', 'gi_typelibs'),

this includes the HarfBuzz typedef file and when I run the installer I now get a version of Zim which works!

The change I made to the build.sh file to use pyinstaller 3.6 was as follows

pip install PyGObject xdg 'pyinstaller==3.6'

Hope this information is of use in any future releases.
